shared/utils/urls.ts

```diff
@@ -54,7 +55,7 @@ export function isUrl(text: string) {
 
   try {
     const url = new URL(text);
-    return url.hostname !== "";
+    return url.hostname !== "" && !blacklist_protocols.includes(url.protocol);
```

Comment:
Summing everything up, since `hostname` will always be an empty string, the `isUrl` function will always return true (which will lead to have the URLs sanitized) in all of the following cases: 

`data:text/html,<script>alert('hi');</script>` 
`vbscript:whatever`
`javascript:whatever`
`vbscript:whatever`

and the protocols check will never be reached
```JS
!blacklist_protocols.includes(url.protocol); 
```

Only in the case of the `file` protocols, the check will save us. But since we're not very sure how malicious URLs might be written and whether they may include hostnames to bypass this or not, I've kept the check. 

If any of you knows a malicious URL with a hostname, we would need to add them to tests in order for this to be really tested, because now, even if we removed the blacklist protocols check, the examples above would still be sanitized because they're not URL according to `isURL` due to the fact that there's no hostname.  

I've created a commit